std/typecons.d

```diff
@@ -746,6 +777,64 @@ template Tuple(Specs...)
         }
 
         /**
+         * Iterate fields in tuple as a _range.
+         *
+         * Only available when all field types are exactly the same in
+         * unqualified form. When the field types are all the same but with
+         * different mutability, the element type of the _range is `const`.
+         *
+         * Not available for the empty tuple.
+         *
+         * Returns:
+         *     Slice of the fields inside this tuple; the lifetime of the
+         *     slice must be no longer than the lifetime of the tuple.
+         */
+        static if (Types.length && !is(CommonLvalueType!Types == void))
+        {
+            CommonLvalueType!Types[] range() @system
+            {
+                return (&expand[0])[0 .. Types.length];
+            }
+
+            ///
+            nothrow unittest
```

Comment:
It's a performance issue: code is generated and executed an unnecessary amount of times. I haven't heard of any measurements indicating that `Tuple` in particular is a problem but it's good to avoid in principle.